ReactSkia/textlayoutmanager/react/renderer/textlayoutmanager/TextLayoutManager.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * Copyright (C) 1994-2021 OpenTV, Inc. and Nagravision S.A.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <memory>
+
+#include <react/renderer/attributedstring/AttributedString.h>
+#include <react/renderer/attributedstring/AttributedStringBox.h>
+#include <react/renderer/attributedstring/ParagraphAttributes.h>
+#include <react/renderer/core/LayoutConstraints.h>
+#include <react/renderer/textlayoutmanager/TextMeasureCache.h>
+#include <react/utils/ContextContainer.h>
+
+#include <RSkTextLayoutManager.h>
+#include <glog/logging.h>
+
+/* Enable to print values of text & paragraph attributes for text computation */
+/*#define RNS_DEBUG_TEXT_ATTRIBUTES*/
+
+namespace facebook {
+namespace react {
+
+class TextLayoutManager;
+
+using SharedTextLayoutManager = std::shared_ptr<const TextLayoutManager>;
+
+/*
+ * Cross platform facade for Skia-specific TextLayoutManager.
+ */
+class TextLayoutManager : public RSkTextLayoutManager {
+ public:
+  TextLayoutManager(const ContextContainer::Shared &contextContainer)
+      : contextContainer_(contextContainer){};
+  ~TextLayoutManager();
+
+  /*
+   * Measures `attributedStringBox` using native text rendering infrastructure.
+   */
+  TextMeasurement measure(
+      AttributedStringBox attributedStringBox,
+      ParagraphAttributes paragraphAttributes,
+      LayoutConstraints layoutConstraints) const;
+
+  /*
+   * Measures lines of `attributedString` using native text rendering
+   * infrastructure.
+   */
+  LinesMeasurements measureLines(
+      AttributedString attributedString,
+      ParagraphAttributes paragraphAttributes,
+      Size size) const;
+
+  /*
+   * Returns an opaque pointer to platform-specific TextLayoutManager.
+   * Is used on a native views layer to delegate text rendering to the manager.
+   */
+  void *getNativeTextLayoutManager() const;
```

Comment:
It seems this and `self_` not being used? any chances to remove this?

Reply:
Yes you are right, its actually not used. Since it was re-used code from ReactCommon cxx platform code, it remained here. It can be removed. Will update the change.